docs/core/transactions.md

```diff
@@ -0,0 +1,84 @@
+
+Note: module messages are not to be confused with [ABCI Messages](https://tendermint.com/docs/spec/abci/abci.html#messages) which define interactions between the Tendermint and application layers. While ABCI messages such as CheckTx and DeliverTx contain Transactions, which contain module Messages, they are not to be confused with the module level messages themselves
+
+The [`Msg`](https://github.com/cosmos/cosmos-sdk/blob/master/types/tx_msg.go#L10-L31) interface has five required functions.
+
+* **Route** returns a string that specifies which module this message is a part of and, thus, which module contains the handler used to implement the actions of this message. [Baseapp](./baseapp.md) uses this function to deliver transactions by invoking the correct handler(s). For example, `MsgSend` is defined in and handled by the `bank` module; its `Route` function returns the name of the `bank` module so `Baseapp` understands to use that module's handler.
+* **Type** returns a short, human-readable string that describes what the message does. For example, `MsgSend`'s type is `"send"`.
+* **ValidateBasic** implements a set of stateless sanity checks for the message and returns an `Error`. For example, the `validateBasic` method for `MsgSend`, the message which sends coins from one account to another, checks that both sender and recipient addresses are valid and the amount is nonnegative.
+* **GetSignBytes** returns a `[]byte` representation of the message which is used by the signer to sign it.
+* **GetSigners** returns a list of addresses whose corresponding signatures are required for the message to be valid. For example, `MsgSend` requires a signature from the sender of the coins.
+
+While messages contain the information for state transition logic, a transaction's other metadata and relevant information are stored in the `TxBuilder` and `CLIContext`.
+
+## Transaction Generation
+
+[`Contexts`](https://godoc.org/context) are immutable objects that contain all the information needed to process a request. In the process of creating a transaction, two contexts are created: the [`CLIContext`](../interfaces/query-lifecycle.md#clicontext) and `TxBuilder`. Both are automatically generated and do not need to be defined by application developers, but do require input from the transaction creator (e.g. using flags through the CLI).
+
+The [`TxBuilder`](https://github.com/cosmos/cosmos-sdk/blob/master/x/auth/types/txbuilder.go) contains data closely related with the processing of transactions:
```

Comment:
Well it's one way of creating transaction (through `auth` module), it hsould be specified that it's not necessarily mandatory to do it this way